canonical/maas-ui
# Feat: - MAASENG-1370 - implement shift click machine selection

## Done

- Select a range of machines by clicking a checkbox, holding `shift` and clicking another checkbox (This works only when there's no grouping).

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to machine list page (`/MAAS/r/machines`)
- Ensure that grouping is set to "No grouping"
- Click on a checkbox to select a machine
- skip a few others above or below the previous checkbox, hold `Shift` key and click on another checkbox

## Fixes

Fixes: # .
[MAASENG-1370](https://warthogs.atlassian.net/browse/MAASENG-1370)
closes #1041 
## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

<img width="1425" alt="image" src="https://user-images.githubusercontent.com/47540149/219316506-0d34dd07-365e-4364-b19d-d5b85c82a1f8.png">


[MAASENG-1370]: https://warthogs.atlassian.net/browse/MAASENG-1370?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
